### PR TITLE
Item Conditions and wave loop

### PR DESCRIPTION
**Pull request by Loerad**
This branch adds the new rarity binding to items that happen each 5 waves. I am also going to add the skip upgrade button, and looping mechanic in this branch. 

The looping mechanic I have decided to add will modulo the wave number with 25 and display the remainder as the wave number. This is so I can use the remainder for the item calculation but still keep the wave number for the enemy spawning calculation

### CODE REVIEW

File: Assets/Scenes/CollisionFix.unity
Comment (PBWiseman) [inline Assets/Scenes/CollisionFix.unity:1]: In the CollisionFix scene the Wave Number has been set to start at 0 which gives an epic update at the start of the game. Just need to make sure this doesn't get moved to the main scene with the other changes.

**Review by PBWiseman — APPROVED** [review]
New upgrades system works well. Other than the issue about starting at wave 0 it all seems good to merge.

File: Assets/Scenes/CollisionFix.unity
Comment (Loerad) [reply Assets/Scenes/CollisionFix.unity:1]: I assume you were meaning the item scene I was using. This won't, its a change to the surface values in timer manager that can be easily fixed/changed.

**Review by AlexReidNZ — APPROVED** [review]
This works and looks great, the weapon selection is really cool. The unimplemented items should probably be given either some temporary functionality or an indication to the player they aren't yet implemented (acceptable since we're still in alpha). Other than that this looks good to merge

**Review by kblack89 — APPROVED** [review]
This works without any apparent issues from testing and looking at the code. Visually looks good as well. This looks good to merge

**Pull request merged**